### ISSUE TITLE
importPreset overwrites Array Object

### ISSUE THREAD

**Issue by chrsmlls333** [cocopon/tweakpane#500]
Hi there, loving this library but I have run into a problem. Was looking forward to the import/export features but I think the import process is assigning a new Array object rather than its elements/properties.

I am using math.gl which has Vector/2/3 objects that extend the built-in Array class, adding useful functions for vector/matrix math. They have an inherent [] store like its superclass, and xyz getter/setters. I detected this issue when the extra subclass functions disappeared and the bound object became a simple array on import.

To ensure the bound Array object and its reference is not lost, I propose that importPreset should set an array's entries to those of the preset instead of overwriting the array itself.

Some minimal test code to see the error:
```js
"dependencies": {
    "@math.gl/core": "3.6.3",
    "tweakpane": "3.1.4"
  }
```
```js
import { Pane } from 'tweakpane';
import { Vector2 } from "@math.gl/core";

var state = {  vector: new Vector2(6, 9) }

console.log(state.vector.constructor.name) // Vector2

var p = new Pane()
p.addInput(state, "vector")

var preset = JSON.stringify(p.exportPreset())

try {
  p.importPreset(JSON.parse(preset))
} catch (error) {
  console.error(error) // TpError {message: "Property 'x' not found"}
}

console.log(state.vector.constructor.name); // Array
```

Thanks for your help on this!

**Comment by cocopon** [comment]
Thank you for reporting the issue.

The bug that `importPreset()` breaks bound values has been fixed in the latest version 3.1.5.

But the code above still doesn't work properly because you uses `JSON.stringify()` and `JSON.parse()` for the preset object.

Steps to occur the problem:

1. `exportPreset()` exports the Vector2 instance: `{vector: Vector2 [6, 9]}`
2. `JSON.stringify()` converts it into a plain array: `{"vector": [6, 9]}`
3. `JSON.parse()` restores it as a plain array: `{vector: [6, 9]}`
4. `importPreset()` fails because it has no key `x` and `y`

If you want to use a stringified preset, you should restore Vector2 instance or a plain object `{x, y}` from a plain array before importing.

**Comment by chrsmlls333** [comment]
Oh that is a total gap I hadn't considered. Thank you!

So even though its an array at heart, tweakpane is binding to Vector3 as `{ xyz }` object because of the getter properties existing. I had assumed that the PresetObject would be sanitized to only pure objects/arrays that tweakpane can understand, in whatever limited schema, not have foreign class instances passed through it. I guess this is a weird edge case. I'll try to create a replacer for when I stringify. I appreciate the guidance.

I'm having a secondary color import issue but I will make a new issue for that.
